core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillWriter.java

```diff
@@ -61,6 +65,9 @@ public UnsafeSorterSpillWriter(
     this.file = spilledFileInfo._2();
     this.blockId = spilledFileInfo._1();
     this.numRecordsToWrite = numRecordsToWrite;
+    this.diskWriteBufferSize =
```

Comment:
I don't have special interest over this, but at least we should keep the same behavior for both `ShuffleExternalSorter` class and `UnsafeSorterSpillWriter` class, otherwise we are introducing unnecessary complexity. Could you please update this?